R/tostring.R

```diff
@@ -826,14 +1074,144 @@ wrap_string <- function(str, width, collapse = NULL) {
 }
 
 .go_stri_wrap <- function(str, w) {
+    if(is.na(w) || is.nan(w))
+        browser()
+  if(w < 1)
+    return(str)
   stringi::stri_wrap(str,
     width = w,
     normalize = FALSE, # keeps spaces
     simplify = TRUE, # If FALSE makes it a list with str elements
-    indent = 0
+    indent = 0,
+    use_length = FALSE # incase the defaul changes, use actual char widths
   )
 }
 
+#' @rdname wrap_string_ttype
+#' @export
+split_word_ttype <- function(str, width, fontspec, min_ok_chars) {
+  chrs <- strsplit(str, "")[[1]]
+  nctt_chars <- nchar_ttype(chrs, fontspec, raw = TRUE)
+  ok <- which(cumsum(nctt_chars) <= width)
+  if(length(ok) < min_ok_chars || length(chrs) - length(ok) < min_ok_chars)
+    list(ok = character(),
+         remainder = str)
+  else
+    list(ok = substr(str, 1, length(ok)),
+         remainder = substr(str, length(ok) + 1, nchar(str)))
+}
+
+## need a separate path here because **the number of characters**
+## in each part is no longer going to be constant the way it
+## was for monospace
+## this is much slower but still shouldn't be a bottleneck, if it is we'll
+## have to do something else
+#' wrap string given a Truetype font
```

Comment:
Could you add here a description of roughly how it works? I think it is important also for me to understand how it works ;)

Reply:
I read it in detail now. I have some question but it is somehow clear ;) we are meeting anyway for details